metis-authentication/metis-authentication-service/src/main/java/eu/europeana/metis/authentication/service/AuthenticationService.java

```diff
@@ -119,21 +126,24 @@ public MetisUser updateUserFromZoho(String email)
     }
 
     psqlMetisUserDao.updateMetisUser(metisUser);
-    return convert(metisUser);
+    return convert(storedMetisUser)
+        .orElseThrow(COULD_NOT_CONVERT_EXCEPTION_SUPPLIER);
```

Comment:
Why not put the `.orElseThrow(...)` in the `convert` method? It is applied every time when that method is called.

Reply:
Sure